WooCommerce/Classes/POS/Presentation/Card Present Payments/PointOfSaleCardPresentPaymentEventPresentationStyle.swift

```diff
@@ -127,20 +127,29 @@ extension CardPresentPaymentEventDetails {
             return .message(.tapSwipeOrInsertCard(
                 viewModel: PointOfSaleCardPresentPaymentTapSwipeInsertCardMessageViewModel(
                     inputMethods: inputMethods)))
-
         case .paymentSuccess:
             return .message(.paymentSuccess(viewModel: PointOfSaleCardPresentPaymentSuccessMessageViewModel()))
         case .paymentError(error: let error, tryAgain: let tryAgain, cancelPayment: _):
-            return .message(.paymentError(
-                viewModel: PointOfSaleCardPresentPaymentErrorMessageViewModel(
-                    error: error,
-                    tryAgainButtonAction: tryAgain)))
-
+            switch error {
+            case CollectOrderPaymentUseCaseError.couldNotRefreshOrder:
+                return .message(.validatingOrderError(viewModel: .init(error: error, tryAgainButtonAction: tryAgain)))
+            default:
+                return .message(.paymentError(
+                    viewModel: PointOfSaleCardPresentPaymentErrorMessageViewModel(
+                        error: error,
+                        tryAgainButtonAction: tryAgain)))
+            }
         case .paymentErrorNonRetryable(error: let error, cancelPayment: _):
-            return .message(.paymentErrorNonRetryable(
-                viewModel: PointOfSaleCardPresentPaymentNonRetryableErrorMessageViewModel(
-                    error: error)))
-
+            switch error {
```

Comment:
@joshheald added handling of additional order validation-related errors. Those are all non-retryable so I'm hiding a retry button:

https://github.com/user-attachments/assets/002a8a80-eadd-4916-bc80-a81444a5223a